db/collection.go

```diff
@@ -341,13 +431,35 @@ func (t *Txn) Delete(ids ...core.InstanceID) error {
 
 // Has returns true if all IDs exists in the collection, false otherwise.
 func (t *Txn) Has(ids ...core.InstanceID) (bool, error) {
+	if err := t.collection.db.connector.Validate(t.token, true); err != nil {
+		return false, err
+	}
+	pk, err := t.token.PubKey()
+	if err != nil {
+		return false, err
+	}
 	for i := range ids {
 		key := baseKey.ChildString(t.collection.name).ChildString(ids[i].String())
 		exists, err := t.collection.db.datastore.Has(key)
 		if err != nil {
 			return false, err
 		}
-		if !exists {
+		if exists {
+			if t.collection.readFilter == nil {
+				continue
+			}
+			bytes, err := t.collection.db.datastore.Get(key)
+			if err != nil {
+				return false, err
+			}
+			bytes, err = t.collection.filterRead(pk, bytes)
+			if err != nil {
+				return false, err
+			}
+			if bytes == nil { // Access denied
+				return false, nil
+			}
+		} else {
 			return false, nil
 		}
```

Comment:
Some general comment now that I see this `Has` method using the read filter.
I wonder if an extra comment in the method might help a bit to understand what this is doing now with the existence of this _optional_ filter.

The `Has` function will return _false_ even if that key exist in the ThreadDB but you're not allowed to see it. So feels like operations are more like _your view about the ThreadsDB_ rather than the DB.
I guess that's the new way of interpreting methods in ThreadsDB, so maybe that comment clarification is unnecessary. Sharing thoughts.

We aren't now in perf considerations, but maybe making read filters accept a batch of instances might alleviate a bit computational things. Also, I'm not sure if JS funcs can be cached to avoid som VM-JS compilation? (not sure if done, just a feeling) So caching+batching _could_ help a bit in the future. (e.g: in this case, grouping `Has` ids in 10, and validating groups). Anyway, out of scope now... brainstorming.

Reply:
Nice, I'll create a couple issues for this... I definitely should have cached the `goja.Runtime` and the compiled `goja.Callable` object on `Collection`.

Reply:
Nice, sounds good.